pages/sdk-protocol-kit/guides/safe-signatures.md

```diff
@@ -0,0 +1,848 @@
+
+It's possible to use several signing methods, such as `ETH_SIGN` (eth_sign), `ETH_SIGN_TYPED_DATA_V4` (eth_signTypedData_v4), ...etc. The default signing method is `ETH_SIGN_TYPED_DATA_V4`.
+
+`ETH_SIGN` produces a regular [EIP-191](https://eips.ethereum.org/EIPS/eip-191) signature. `ETH_SIGN_TYPED_DATA_V4` produces a signature that's compatible with the [EIP-712](https://eips.ethereum.org/EIPS/eip-712) standard. The EIP-712 standard is used by Safe contracts to verify the signatures.
+
+```typescript
+safeTx = await protocolKit.signTransaction(safeTx, SigningMethod.ETH_SIGN); // EIP-191 signature from owner1
+protocolKit = await protocolKit.connect({ ethAdapter: ethAdapter2 }); // Connect another owner
+safeTx = await protocolKit.signTransaction(
+  safeTx,
+  SigningMethod.ETH_SIGN_TYPED_DATA_V4
+); // owner2 EIP-712 typed data signature
+```
+
+In this snippet, we add the signature for `owner1`. Then, we use the `connect()` method to connect the `owner2` adapter and sign again to add the second signature.
+
+If we examine the `safeTx` object at this point, we should see something similar to the following:
+
+```javascript
+EthSafeTransaction {
+  signatures: Map(2) {
+    '0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1' => EthSafeSignature {
+      signer: '0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1',
+      data: '0x969308e2abeda61a0c9c41b3c615012f50dd7456ca76ea39a18e3b975abeb67f275b07810dd59fc928f3f9103e52557c1578c7c5c171ffc983afa5306466b1261f',
+      isContractSignature: false
+    },
+    '0xffcf8fdee72ac11b5c542428b35eef5769c409f0' => EthSafeSignature {
+      signer: '0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0',
+      data: '0x4d63c79cf9d743782bc31ad58c1a316020b39839ab164caee7ecac9829f685cc44ec0d066a5dfe646b2ffeeb37575df131daf9c96ced41b8c7c4aea8dc5461801c',
+      isContractSignature: false
+    }
+  },
+  data: { ... }
+}
+```
+
+The `data` prop in the `signatures` map represents the concrete signature. The `isContractSignature` flag (_false_) indicates if the signature is an Ethereum signature or a Contract signature (A signer Safe Account).
+
+An Ethereum signature is composed of two 32-byte integers (r, s) and an extra byte for recovery id (v), making a total of 65 bytes. In hexadecimal string format, each byte is represented by 2 characters. Hence, a 65-byte Ethereum signature will be 130 characters long. Including the '0x' prefix commonly used with signatures, the total character count for such a signature would be 132. For a more detailed explanation, you can refer to [this link](https://docs.safe.global/safe-smart-account/signatures) for more information.
+
+> To represent a byte (8 bits) in hexadecimal, you need 2 characters. Each hexadecimal character represents 4 bits. Therefore, 2 hexadecimal characters (2 x 4 bits) are able to represent a byte (8 bits).
+
+The final part of the signature, either `1f` or `1c`, indicates the signature type.
```

Comment:
There are the following V values supported by Safe:
- 0 (contract signature)
- 1 (approved hash)
- ethereum-adjusted ECDSA recovery byte {27, 28} + 4 for eip-191 signed message
- ethereum-adjusted ECDSA recovery byte for straightforward ecdsa sig

More on this here: https://ethereum.stackexchange.com/questions/113499/why-is-value-of-ecdsa-v-often-neither-27-nor-28

It's important that for the EIP-191 signed message the V is adjusted to the ecdsa V + 4. If the generated value is 28 and you adjust it to `0x1f`, the signature verification will fail, it should be `0x20` (28+4=32) instead